### PR TITLE
feat(apache): migrate to ingest

### PR DESCRIPTION
**Pull request by PierrickV**
| Before the update                                                            | After the update                    |
| ---------------------------------------------------------------------------- | ----------------------------------- |
| (no impact expected) at the parser level `action.properties` is of type list | `action.properties` is of type dict |

Parent task: [#41572](https://github.com/SekoiaLab/platform/issues/41572)

### CODE REVIEW

File: Apache/apache/tests/access_combined.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      "created": "2020-06-12T09:43:30.348",
+      "id": "10f0afe9-98a1-4226-a6bd-8f70d461d430"
+    },
+    "log": {
+      "hostname": "B96f1GJTxDUKbh2l"
+    },
+    "message": "127.0.0.1 - userfoo [10/Oct/2000:13:55:36 -0700] \"GET /apache_pb.gif HTTP/1.0\" 200 2326 \"http://www.example.com/start.html\" \"Mozilla/4.08 [en] (Win98; I ;Nav)\""
+  },
+  "expected": {
+    "message": "127.0.0.1 - userfoo [10/Oct/2000:13:55:36 -0700] \"GET /apache_pb.gif HTTP/1.0\" 200 2326 \"http://www.example.com/start.html\" \"Mozilla/4.08 [en] (Win98; I ;Nav)\"",
```
Comment (TOUFIKIzakarya) [inline Apache/apache/tests/access_combined.json:13]: Why we don't parse the `timestamp` ?

File: Apache/apache/tests/error.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      "created": "2020-06-12T09:43:30.348",
+      "id": "10f0afe9-98a1-4226-a6bd-8f70d461d430"
+    },
+    "log": {
+      "hostname": "B96f1GJTxDUKbh2l"
+    },
+    "message": "[Wed Oct 11 14:32:52 2000] [error] [client 127.0.0.1] client denied by server configuration: /export/home/live/ap/htdocs/test"
+  },
+  "expected": {
+    "message": "[Wed Oct 11 14:32:52 2000] [error] [client 127.0.0.1] client denied by server configuration: /export/home/live/ap/htdocs/test",
```
Comment (TOUFIKIzakarya) [inline Apache/apache/tests/error.json:13]: Why we don't parse the `timestamp` ?

**Review by TOUFIKIzakarya — COMMENTED** [review]
Hi @PierrickV, thanks for the job. 
I think that it will be awesome to add some more tests if it's possible.

**Comment by PierrickV** [comment]
@TOUFIKIzakarya: Thank you for taking the time to review my work :pray: 

We do not parse the `timestamp` because the previous parser did not. The migration does not aim at improving the parser it is just about changing the underlining technology. We want to be able to deploy the exact same behavior to production. The improvements will be carried by the @SEKOIA-IO/sekoia-io-integration-team once the migration is finished.

If you see that the `timestamp` was present in the previous logs it is due to an unrelated behavior related to Syslog (same applies for `host.name` & `host.hostname`).

Have a great day :sun_with_face:

**Comment by PierrickV** [comment]
Hello @TOUFIKIzakarya 

I added some tests and a fix for the parsing of `HTTPD_COMMONLOG`

File: Apache/apache/ingest/parser.yml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      - set:
+          action.properties.matchoffset: "{{grok.event.matchoffset}}"
+          action.properties.modsecmessage: "{{grok.event.modsecmessage}}"
+          action.properties.ruledata: "{{grok.event.ruledata}}"
+          action.properties.rulefile: "{{grok.event.rulefile}}"
+          action.properties.ruleid: "{{grok.event.ruleid}}"
+          action.properties.ruleline: "{{grok.event.ruleline}}"
+          action.properties.rulerev: "{{grok.event.rulerev}}"
+          action.properties.ruleseverity: "{{grok.event.ruleseverity}}"
+          action.properties.timestamp: "{{grok.event.timestamp}}"
```
Comment (squioc) [inline Apache/apache/ingest/parser.yml:52]: Why not parse the timestamp and use it into `@timestamp`?

**Review by squioc — COMMENTED** [review]
LGTM.

File: Apache/apache/ingest/parser.yml (hunk 3 above)
Comment (squioc) [reply Apache/apache/ingest/parser.yml:52]: I got the answer.

**Review by squioc — APPROVED** [review]

**Review by TOUFIKIzakarya — APPROVED** [review]

**Review by squioc — APPROVED** [review]

**Review by squioc — APPROVED** [review]

**Pull request merged**